### ISSUE TITLE
pkgdown cannot be loadNamespace()d

### ISSUE THREAD

**Issue by maxheld83** [r-lib/ghactions#247]
all of the dev helpers so far could be just installed to `R_LIBS_ACTION`, and be loaded right before they are needed, *without* prepending the whole `R_LIBS_ACTION` to `.libPaths()`. 

This does not appear to work for pkgdown.

```
loadNamespace(package = "pkgdown", lib.loc = Sys.getenv("R_LIBS_ACTION"))
```

runs without a hitch, but, say `pkgdown::build_site()` then still fails.

This can also be reproduced interactively via `docker run -it --entrypoint=bash maxheld83/pkgdown`:

```
> loadNamespace(package = "pkgdown", lib.loc = Sys.getenv("R_LIBS_ACTION"))
<environment: namespace:pkgdown>
> pkgdown::build_site()
Error in loadNamespace(name) : there is no package called ‘pkgdown’
Error in loadNamespace(name) : there is no package called ‘pkgdown’
```

If you first set `R_LIBS="$R_LIBS_ACTION"` the whole thing works.

So it seems that there is something particularly wrong with `loadNamespace()` for pkgdown.

Not a priority, the hack-fix is ok, but would be nice to get to the bottom of this.

**Comment by maxheld83** [comment]
aha!
*pkgdown* uses *callr* [https://github.com/r-lib/pkgdown/blob/bc5dc6eeef26336220d6e496e9f6e5d606938318/R/build.r#L337-L345](internally):

```
callr::r(
    function(..., crayon_enabled, crayon_colors, pkgdown_internet) {
      options(
        crayon.enabled = crayon_enabled,
        crayon.colors = crayon_colors,
        pkgdown.internet = pkgdown_internet
      )
      pkgdown::build_site(...)
    }
```

this makes our usual `loadNamespace()` strategy ineffective, because that does not persist into a new R session.

I'm guessing it'll be the same in covr.

So our strategy is pretty limited, does not work for these kinds of complicated packages.

Will fix by just wrapping it with withr.

**Comment by maxheld83** [comment]
so this actually cannot be "fixed" in ghactions.

**Comment by maxheld83** [comment]
will be closed in #272, because `loadNamespace()` is a stupid idea to begin with.
